ql/src/java/org/apache/hadoop/hive/ql/parse/LoadSemanticAnalyzer.java

```diff
@@ -294,6 +295,14 @@ private void analyzeLoad(ASTNode ast) throws SemanticException {
       throw new SemanticException(ErrorMsg.DML_AGAINST_VIEW.getMsg());
     }
     if (ts.tableHandle.isNonNative()) {
+      // launch a tez job
+      StorageFormatDescriptor ss = ts.tableHandle.getStorageHandler().supportsLoadData(ts.tableHandle.getTTable());
+      if (ss != null) {
```

Comment:
could we just call `StorageFormat.processStorageFormat` that should set inputFormat and serde?